test/e2e/test_job_submissions.py

```diff
@@ -1289,6 +1290,163 @@ def test_worker_uses_job_attachment_configuration(
         finally:
             os.remove(os.path.join(list(output_path.keys())[0], "output_file"))
 
+    def test_job_submission_fails_when_input_file_not_found_and_require_paths_exist(
+        self, deadline_resources: DeadlineResources, deadline_client: DeadlineClient
+    ) -> None:
+        # Tests that when a file specified in the job template is not found, the worker handles the missing file and fails the job creation
+        job_bundle_path: str = os.path.join(
+            os.path.dirname(__file__),
+            "job_attachment_bundle",
+        )
+        job_parameters: List[Dict[str, str]] = [
+            {
+                "name": "NonExistentInputFile",
+                "value": os.path.join(job_bundle_path, "non_existent_file"),
+            },
+        ]
+
+        try:
+            with open(os.path.join(job_bundle_path, "template.json"), "w+") as template_file:
+                template_file.write(
+                    json.dumps(
+                        {
+                            "specificationVersion": "jobtemplate-2023-09",
+                            "name": "MissingFileJob",
+                            "parameterDefinitions": [
+                                {
+                                    "name": "NonExistentInputFile",
+                                    "type": "PATH",
+                                    "objectType": "FILE",
+                                    "dataFlow": "IN",
+                                },
+                            ],
+                            "steps": [
+                                {
+                                    "name": "MainStep",
+                                    "hostRequirements": {
+                                        "attributes": [
+                                            {
+                                                "name": "attr.worker.os.family",
+                                                "allOf": [os.environ["OPERATING_SYSTEM"]],
+                                            }
+                                        ]
+                                    },
+                                    "script": {
+                                        "actions": {"onRun": {"command": "whoami"}},
+                                    },
+                                }
+                            ],
+                        }
+                    )
+                )
+
+            config = configparser.ConfigParser()
+
+            set_setting("defaults.farm_id", deadline_resources.farm.id, config)
+            set_setting("defaults.queue_id", deadline_resources.queue_a.id, config)
+            with pytest.raises(Exception):
```

Comment:
Isn't this just testing the API? What are we testing for the worker here?

Reply:
same question

Reply:
There's no testing for the worker in this case. It tests the job attachments instead so I think I will remove this test. Thanks for calling this out!